Util/Docker/README.md

```diff
@@ -1,104 +1,205 @@
 
-**The Docker images produced by the ue4-docker Python package contain the UE4 Engine Tools in both source code and object code form. As per Section 1A of the [Unreal Engine EULA](https://www.unrealengine.com/en-US/eula), Engine Licensees are prohibited from public distribution of the Engine Tools unless such distribution takes place via the Unreal Marketplace or a fork of the Epic Games UE4 GitHub repository. Public distribution of the built images via an openly accessible Docker Registry (e.g. Docker Hub) is a direct violation of the license terms. It is your responsibility to ensure that any private distribution to other Engine Licensees (such as via an organization's internal Docker Registry) complies with the terms of the Unreal Engine EULA.**  
+### OS Setup
+Install the latest updates.
 
-For more details, see the [Unreal Engine EULA Restrictions](https://unrealcontainers.com/docs/obtaining-images/eula-restrictions) page on the [Unreal Containers community hub](https://unrealcontainers.com/).
+```sh
+sudo apt-get update && sudo apt-get upgrade -y
+reboot
+```
 
-## Requirements
+### Install Docker with NVIDIA GPU Support
 
-```
-- 64-bit version of Docker in Ubuntu 16.04+.
-- Minimum 8GB of RAM
-- Minimum 300GB available disk space for building container images
-```
+1) Choose a **proprietary** NVIDIA driver (don't waste time with the open source one!!!)
 
----
+![](./nvidia-driver-selection.png)
 
-## Prerequisites
+2) Execute following commands to install Docker with GPU support
 
-You need Docker installed and configured so your Docker images can access to the Internet during the build process.
+```sh
+# install cURL, Docker and Docker-Compose
+sudo apt-get update && sudo apt-get install -y curl docker.io docker-compose
 
-Make sure you have installed **Python 3.6 or newer**, check that is in the path, and is callable using `python3` in your terminal. One possible way to achieve so is by using [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/).
+# register the NVIDIA Docker PPA as apt source
+distribution=$(. /etc/os-release;echo $ID$VERSION_ID) \
+    && curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add - \
+    && curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list \
+        | sudo tee /etc/apt/sources.list.d/nvidia-docker.list
 
-## Dependencies
+# install NVIDIA Docker from the official NVIDIA PPA
+sudo apt-get update && sudo apt-get install -y nvidia-docker2
 
-As mentioned before, in order to use Unreal Engine inside Docker we use [ue4-docker](https://github.com/adamrehn/ue4-docker). You can install it using `pip3`.  
-Further information on installing ue4-docker on Linux can be found [here](https://adamrehn.com/docs/ue4-docker/configuration/configuring-linux).
+# disable the swap partition. be smart, don't kill your disk :)
+sudo sed -i '/ swap / s/^\(.*\)$/#\1/g' /etc/fstab && sudo mount -a
```

Comment:
Can you explain to me what happens here?

I thought swap is a dedicated space on your ROM to provide additional space for your RAM, if RAM is full.

How does this kill my disk?

Reply:
Ehm yeah. Sed is a pattern replacement tool. It searches for lines containing "swap" and puts a commenting '#' at the start of each matching line to comment it.

This is important because otherwise overflowing RAM would be dealt with by caching stuff to disk (which is btw not performing very well either). As with today's SSD devices, you can quickly kill your disk in such a case. So, every reasonable Docker deployment should disable swap. In case a container overflows, it may just crash an restart or something like that. Much cleaner and doesn't kill your hardware.